### PR TITLE
Polish auth xmod

### PR DESCRIPTION
**Pull request by NYBACHOK**

### CODE REVIEW

File: gears/src/baseapp/genesis.rs [hunk 1]
```diff
@@ -8,6 +8,7 @@ pub struct GenesisError(pub AccAddress);
 pub trait Genesis:
     std::fmt::Debug + Default + DeserializeOwned + Serialize + Clone + Send + Sync + 'static
 {
+    // TODO: REMOVE?
```
Comment (NYBACHOK) [inline gears/src/baseapp/genesis.rs:11]: Is there any way to rework this? As I see this method needed by 2 xmods only.

File: x/auth/src/keeper.rs [hunk 2]
```diff
@@ -144,118 +139,65 @@ impl<SK: StoreKey, PSK: ParamsSubspaceKey, M: Module> Keeper<SK, PSK, M> {
         }
     }
 
-    pub fn init_genesis<DB: Database>(
+    pub fn init<DB: Database, CTX: TransactionalContext<DB, SK>>(
```
Comment (joneskm) [inline x/auth/src/keeper.rs:142]: Why `TransactionalContext` vs `InitContext`?

Comment (NYBACHOK) [reply x/auth/src/keeper.rs:142]: I split single method into 2 parts and wanted to make keeper as more generic as possible

Comment (NYBACHOK) [reply x/auth/src/keeper.rs:142]: As you see, I even removed `_genesis` part from name

File: Makefile [hunk 3]
```diff
@@ -8,7 +8,7 @@ run:
 	cargo run -- run --min-gas-prices 0uatom
 
 test:
-	cargo test --features=macros_test,it  --no-fail-fast 
+	RUST_BACKTRACE=1 cargo test --features=macros_test,it  --no-fail-fast
```
Comment (dmvict) [inline Makefile:11]: I don't think that it's necessary. You may replace it by environment from the command.

**Review by dmvict — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (NYBACHOK) [reply Makefile:11]: True, but on the other hand if run test you want to see where error occurred without re-running them.

File: x/auth/src/keeper.rs (hunk 2 above)
Comment (NYBACHOK) [reply x/auth/src/keeper.rs:142]: I can't say `yes` or `no`. But probably there wouldn't be such cases

Comment (joneskm) [reply x/auth/src/keeper.rs:142]: > ...make keeper as more generic as possible
I generally aim for the exact opposite :wink:

Comment (NYBACHOK) [reply x/auth/src/keeper.rs:142]: Okay. I revert some changes

Comment (NYBACHOK) [reply x/auth/src/keeper.rs]: Edited to use InitContext and removed gas

**Pull request merged**